python/cudf_kafka/cudf_kafka/_lib/kafka.pyx

```diff
@@ -14,7 +14,12 @@ from cudf_kafka._lib.kafka cimport kafka_consumer
 # To avoid including <python.h> in libcudf_kafka
 # we introduce this wrapper in Cython
 cdef map[string, string] oauth_callback_wrapper(void *ctx):
-    return (<object>(ctx))()
+    resp = (<object>(ctx))()
+    cdef map[string, string] c_resp
+    c_resp[str.encode("token")] = str.encode(resp["token"])
+    c_resp[str.encode("token_expiration_in_epoch")] \
+        = str(resp["token_expiration_in_epoch"]).encode()
+    return c_resp
```

Comment:
Do we need to store `token_expiration_in_epoch` as a string? Are there many parameters and we just assume they're all in string. It seems like this would be a numerical value like a `float` or a `long`.

Reply:
This is just the intermediary object transfer. And it makes it easier to not force the Python callback author to use a specific type. The string is converted to a `int64_t` in `kafka_callback.cpp`